eulerpi/core/plotting.py

```diff
@@ -0,0 +1,393 @@
+    color4 = np.array([255.0, 218.0, 174.0]) / 255.0
+
+    # define the locations and extends of the violin plots on the abscissa
+    unit_locations = np.linspace(1, 2 * dim - 1, dim) / (2.0 * dim)
+    envelope_width = 1.0 / (dim + 1)
+
+    # load the results from the inference sampling
+    rm = ResultManager(model_name=model.name, run_name=run_name)
+
+    (
+        reconstructed_param_sample,
+        reconstructed_data_sample,
+        _,
+    ) = rm.load_slice_inference_results(
+        slice=np.linspace(0, model.param_dim - 1, model.param_dim, dtype=int)
+    )
+
+    if what_to_plot == "param":
+        reconstructed_sample = reconstructed_param_sample
+    elif what_to_plot == "data":
+        reconstructed_sample = reconstructed_data_sample
+
+    # determine upper and lower plot bounds according to the credibility levels and what to plot
+    if true_bool == 1:
+        true_sample = np.loadtxt(
+            "Data/"
+            + model.name
+            + "/true_"
+            + str(what_to_plot)
+            + "_sample.csv",
+            delimiter=",",
+        )
+
+        true_data_upper_percentile = np.percentile(
+            true_sample, 100.0 * credibility_level, axis=0
+        )
+        true_data_lower_percentile = np.percentile(
+            true_sample, 100.0 * (1 - credibility_level), axis=0
+        )
+
+        max_val = np.amax(true_data_upper_percentile)
+        min_val = np.amin(true_data_lower_percentile)
+
+    else:
+        reconstructed_data_upper_percentile = np.percentile(
+            reconstructed_sample, 100.0 * credibility_level, axis=0
+        )
+        reconstructed_data_lower_percentile = np.percentile(
+            reconstructed_sample, 100.0 * (1 - credibility_level), axis=0
+        )
+
+        max_val = np.amax(reconstructed_data_upper_percentile)
+        min_val = np.amin(reconstructed_data_lower_percentile)
+
+    # create single figure with variable width
+    fig, ax = plt.subplots(figsize=(2 * dim, 6))
+
+    # set the plot axis limits and labels
+    ax.set_ylim(min_val, max_val)
+    x_labels = axis_labels or [
+        r"$\mathcal{" + variable_name + "}_{" + str(d + 1) + "}$"
+        for d in range(dim)
+    ]
+    ax.set_xticks(unit_locations, x_labels)
+
+    # plot the vertical axes for all violin plots
+    ax.vlines(unit_locations, min_val, max_val, color=color3, linewidth=1.0)
+
+    # create one shared grid for all KDEs
+    vertical_grid = np.transpose(
+        np.array([np.linspace(min_val, max_val, num_vertical_grid_points)])
+    )
+
+    # also create a 1d array for the param grid
+    vertical_grid_1d = np.squeeze(np.asarray(vertical_grid))
+
+    # loop over all dimensions of either the param or the data
+    for i in range(dim):
+
+        reconstructed_matrix = np.transpose(
+            np.array([reconstructed_sample[:, i]])
+        )
+
+        # in case there is a reference for the plotted qunatity avaialbe
+        if true_bool == 1:

```

Comment:
Following our discussion, I would propose to use the bool variable only internally and offer an interface similar to the inference function to the user.
This would mean to allow the user to pass something like `true_samples: Optional[Union[str, os.PathLike, np.ndarray]] = None`. Then we can use the loading snippet from the `inference` function and define `reference_available` from the value of `what_to_plot` and whether `true_samples` is None.